int32 botID
---
bool success